specification/communication/data-plane/CallingServer/preview/2021-06-15-preview/communicationservicescallingserver.json

```diff
@@ -0,0 +1,1467 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationIdentifier"
+          }
+        },
+        "source": {
+          "$ref": "#/definitions/CommunicationIdentifier",
+          "description": "The source of the call."
+        },
+        "subject": {
+          "description": "The subject.",
+          "type": "string"
+        },
+        "callbackUri": {
+          "description": "The callback URI.",
+          "type": "string"
+        },
+        "requestedMediaTypes": {
+          "description": "The requested modalities.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CallModality"
+          }
+        },
+        "requestedCallEvents": {
+          "description": "The requested call events to subscribe to.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EventSubscriptionType"
+          }
+        }
+      }
+    },
+    "PhoneNumberIdentifier": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/PhoneNumberIdentifierModel"
+    },
+    "CommunicationIdentifier": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationIdentifierModel"
+    },
+    "CommunicationUserIdentifier": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationUserIdentifierModel"
+    },
+    "CommunicationCloudEnvironment": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationCloudEnvironmentModel"
+    },
+    "MicrosoftTeamsUserIdentifier": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/MicrosoftTeamsUserIdentifierModel"
+    },
+    "CreateCallResult": {
+      "description": "The response payload of the create call operation.",
+      "type": "object",
+      "properties": {
+        "callConnectionId": {
+          "description": "The call connection id.",
+          "type": "string"
+        }
+      }
+    },
+    "CommunicationError": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationError"
+    },
+    "PlayAudioRequest": {
+      "description": "The request payload for playing audio.",
+      "type": "object",
+      "properties": {
+        "audioFileUri": {
+          "description": "The media resource uri of the play audio request. \r\nCurrently only Wave file (.wav) format audio prompts are supported.\r\nMore specifically, the audio content in the wave file must be mono (single-channel),\r\n16-bit samples with a 16,000 (16KHz) sampling rate.",
+          "type": "string"
+        },
+        "loop": {
+          "description": "The flag indicating whether audio file needs to be played in loop or not.",
+          "type": "boolean"
+        },
+        "operationContext": {
+          "description": "The value to identify context of the operation.",
+          "type": "string"
+        },
+        "audioFileId": {
```

Comment:
Why do you need this? The `audioFileUri` should identify the file uniquely, correct? And if you want to enable caching, you can also use conditional requests/if-none-match/if-modified-since to only get the file if it has changed.

Reply:
audioFileId is used for caching. The service uses the cached file if the same audioFileId is used. Otherwise it will download it again. We can consider alternatives to this post-public preview

Reply:
It is ok to do this after the public preview. I would recommend renaming the field, however. There is nothing in the name `audioFieldId` that hints at what it is being used for. I suspect that the name had bled out from the internal implementation/the name of the field in the data store you are using.